sound/soc/sof/intel/hda.c

```diff
@@ -720,6 +720,8 @@ int hda_dsp_probe(struct snd_sof_dev *sdev)
 	pci_free_irq_vectors(pci);
 free_streams:
 	hda_dsp_stream_free(sdev);
+hdac_bus_unmap:
+	iounmap(bus->remap_addr);
 err:
```

Comment:
if (bus->remap_addr)
		iounmap(bus->remap_addr);

And when looking into hda_init() more, only pci_ioremap_bar() may get error and return the error(hda_dsp_ctrl_get_caps() returns 0 always, maybe we should change its return type to void also), so we may don't need this iounmap() here, and do iounmap() in hda_dsp_remove().